python/lsst/daf/butler/registry/datasets/byDimensions/_manager.py

```diff
@@ -342,8 +334,15 @@ def resolve_wildcard(
     ) -> dict[DatasetType, list[str | None]]:
         wildcard = DatasetTypeWildcard.from_expression(expression)
         result: defaultdict[DatasetType, set[str | None]] = defaultdict(set)
+        deprecation_message = (
+            "Querying for component datasets via Registry query methods is deprecated in favor of using "
+            "DatasetRef and DatasetType methods on parent datasets. Only components=False will be supported "
+            "after v26."
```

Comment:
Maybe clarify that the parameter will be removed completely after v27 (which is a long time from now). We will need a jira ticket linked to a release somehow.

Reply:
I've created DM-36457 for the post-v27 removals, but haven't tried to create release tickets for v26 or v27 (neither exist yet).